File: jetpack-udemy/Dependency/Dagger2Singleton/app/src/main/java/com/androidtutz/anushka/didemo/NCBatteryModule.java
package com.androidtutz.anushka.didemo;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module
abstract class NCBatteryModule {

    @Binds
    abstract Battery bindNCBattery(NickelCadmiumBattery nickelCadmiumBattery);
}

File: jetpack-udemy/Dependency/Singletons/app/src/main/java/com/anushka/didemo/NCBatteryModule.kt
package com.anushka.didemo

import dagger.Binds
import dagger.Module
import dagger.Provides

@Module
abstract class NCBatteryModule {
    @Binds
    abstract fun bindsNCBattery(nickelCadmiumBattery: NickelCadmiumBattery):Battery
}